frontend/app/components/mapbox/carto-vector-source.js

```diff
@@ -0,0 +1,108 @@
+import Component from '@ember/component';
+import fetch from 'fetch';
+import { action, computed } from '@ember-decorators/object';
+
+export const CARTO_USERNAME = 'planninglabs';
+export const CARTO_MAPS_ENDPOINT = `https://${CARTO_USERNAME}.carto.com/api/v1/map`;
+
+/**
+ *
+ * MapboxCartoLayersComponent renders vector tile layers via e-mapbox-gl.
+ * It provides a contextual component necessary for setting up layers:
+ *
+ * ```
+ * {{#mapbox/carto-vector-source map=map as |carto-layer|}}
+ *   {{carto-layer
+ *     sql='select * from geodata'
+ *     layer=(hash fill-color='blue')
+ *   }}
+ * {{/mapbox/carto-vector-source}}
+ * ```
+ *
+ * Currently this can't handle subsequent invocations of carto-layers because
+ * of a limitation with the ember-mapbox-gl bindings lib and there isn't an
+ * API for it. See https://github.com/mapbox/mapbox-gl-js/pull/8048.
+ */
+export default class MapboxCartoLayersComponent extends Component {
```

Comment:
Is the component class naming style based on what the component yields? Curious how I can use this naming strategy to think about code (I'm used to the mundanely matching the class name to the file name)